[
    {
        "user": "U010GM77S4W",
        "type": "message",
        "ts": "1585644804.010000",
        "client_msg_id": "f5e3604d-6e53-4855-8549-2bde0874ebff",
        "text": "So im trying to run the example marketrecorder.py but I get the following error:\nTraceback (most recent call last):\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\Lib\\site-packages\\flumine\\examples\\marketrecorder.py\", line 46, in &lt;module&gt;\n    framework.add_strategy(strategy)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\flumine\\baseflumine.py\", line 66, in add_strategy\n    self.strategies(strategy)  # store in strategies\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\flumine\\strategy\\strategy.py\", line 26, in _call_\n    strategy.add()\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\Lib\\site-packages\\flumine\\examples\\strategies\\marketrecorder.py\", line 154, in add\n    super().add()\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\Lib\\site-packages\\flumine\\examples\\strategies\\marketrecorder.py\", line 37, in add\n    raise OSError(\"File dir %s does not exist\" % self.local_dir)\nOSError: File dir \/tmp does not exist\n\nI tried to create the dir in question but get the same. What am I missing? Appologies for the dumb question, I'm new to python and coding and am still trying to understand how this library works.",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "66cae8f5399c",
            "image_72": "https:\/\/avatars.slack-edge.com\/2020-03-20\/1004508356769_66cae8f5399cf91a07ae_72.jpg",
            "first_name": "",
            "real_name": "mandelbot",
            "display_name": "mandelbot",
            "team": "T4G9NBD2M",
            "name": "demhadbale",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "3xQ",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "So im trying to run the example marketrecorder.py but I get the following error:\nTraceback (most recent call last):\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\Lib\\site-packages\\flumine\\examples\\marketrecorder.py\", line 46, in <module>\n    framework.add_strategy(strategy)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\flumine\\baseflumine.py\", line 66, in add_strategy\n    self.strategies(strategy)  # store in strategies\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\flumine\\strategy\\strategy.py\", line 26, in "
                            },
                            {
                                "type": "text",
                                "text": "call",
                                "style": {
                                    "italic": true
                                }
                            },
                            {
                                "type": "text",
                                "text": "\n    strategy.add()\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\Lib\\site-packages\\flumine\\examples\\strategies\\marketrecorder.py\", line 154, in add\n    super().add()\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\Lib\\site-packages\\flumine\\examples\\strategies\\marketrecorder.py\", line 37, in add\n    raise OSError(\"File dir %s does not exist\" % self.local_dir)\nOSError: File dir \/tmp does not exist\n\nI tried to create the dir in question but get the same. What am I missing? Appologies for the dumb question, I'm new to python and coding and am still trying to understand how this library works."
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1585644936.010300",
        "edited": {
            "user": "UBS7QANF3",
            "ts": "1585645008.000000"
        },
        "client_msg_id": "31c4a94f-ca38-4649-a6af-495a99d5586e",
        "text": "Change the `local_dir` on this line:  <https:\/\/github.com\/liampauling\/flumine\/blob\/2a261e27d5738ed405f9f1d24446bdfe8b078cc1\/examples\/marketrecorder.py#L35> to somewhere that will exist on Windows",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "JVz=",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Change the "
                            },
                            {
                                "type": "text",
                                "text": "local_dir",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": " on this line:  "
                            },
                            {
                                "type": "link",
                                "url": "https:\/\/github.com\/liampauling\/flumine\/blob\/2a261e27d5738ed405f9f1d24446bdfe8b078cc1\/examples\/marketrecorder.py#L35"
                            },
                            {
                                "type": "text",
                                "text": " to somewhere that will exist on Windows"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "U010GM77S4W",
        "type": "message",
        "ts": "1585646228.010900",
        "edited": {
            "user": "U010GM77S4W",
            "ts": "1585646319.000000"
        },
        "client_msg_id": "0f293514-5f8b-4488-94d5-8e151a57f321",
        "text": "```Doh! Thanks, that was dumb. Although now I'm getting these: Traceback (most recent call last):\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\Lib\\site-packages\\flumine\\examples\\marketrecorder.py\", line 46, in &lt;module&gt;\n    framework.add_strategy(strategy)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\flumine\\baseflumine.py\", line 66, in add_strategy\n    self.strategies(strategy)  # store in strategies\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\flumine\\strategy\\strategy.py\", line 26, in _call_\n    strategy.add()\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\Lib\\site-packages\\flumine\\examples\\strategies\\marketrecorder.py\", line 155, in add\n    self.s3.head_bucket(Bucket=self._bucket)  # validate bucket\/access\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\client.py\", line 316, in _api_call\n    return self._make_api_call(operation_name, kwargs)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\client.py\", line 612, in _make_api_call\n    http, parsed_response = self._make_request(\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\client.py\", line 632, in _make_request\n    return self._endpoint.make_request(operation_model, request_dict)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\endpoint.py\", line 102, in make_request\n    return self._send_request(request_dict, operation_model)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\endpoint.py\", line 132, in _send_request\n    request = self.create_request(request_dict, operation_model)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\endpoint.py\", line 115, in create_request\n    self._event_emitter.emit(event_name, request=request,\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\hooks.py\", line 356, in emit\n    return self._emitter.emit(aliased_event_name, **kwargs)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\hooks.py\", line 228, in emit\n    return self._emit(event_name, kwargs)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\hooks.py\", line 211, in _emit\n    response = handler(**kwargs)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\signers.py\", line 90, in handler\n    return self.sign(operation_name, request)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\signers.py\", line 160, in sign\n    auth.add_auth(request)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\auth.py\", line 357, in add_auth\n    raise NoCredentialsError\nbotocore.exceptions.NoCredentialsError: Unable to locate credentials```",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "66cae8f5399c",
            "image_72": "https:\/\/avatars.slack-edge.com\/2020-03-20\/1004508356769_66cae8f5399cf91a07ae_72.jpg",
            "first_name": "",
            "real_name": "mandelbot",
            "display_name": "mandelbot",
            "team": "T4G9NBD2M",
            "name": "demhadbale",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "KDWA7",
                "elements": [
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Doh! Thanks, that was dumb. Although now I'm getting these: Traceback (most recent call last):\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\Lib\\site-packages\\flumine\\examples\\marketrecorder.py\", line 46, in <module>\n    framework.add_strategy(strategy)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\flumine\\baseflumine.py\", line 66, in add_strategy\n    self.strategies(strategy)  # store in strategies\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\flumine\\strategy\\strategy.py\", line 26, in _call_\n    strategy.add()\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\Lib\\site-packages\\flumine\\examples\\strategies\\marketrecorder.py\", line 155, in add\n    self.s3.head_bucket(Bucket=self._bucket)  # validate bucket\/access\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\client.py\", line 316, in _api_call\n    return self._make_api_call(operation_name, kwargs)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\client.py\", line 612, in _make_api_call\n    http, parsed_response = self._make_request(\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\client.py\", line 632, in _make_request\n    return self._endpoint.make_request(operation_model, request_dict)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\endpoint.py\", line 102, in make_request\n    return self._send_request(request_dict, operation_model)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\endpoint.py\", line 132, in _send_request\n    request = self.create_request(request_dict, operation_model)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\endpoint.py\", line 115, in create_request\n    self._event_emitter.emit(event_name, request=request,\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\hooks.py\", line 356, in emit\n    return self._emitter.emit(aliased_event_name, **kwargs)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\hooks.py\", line 228, in emit\n    return self._emit(event_name, kwargs)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\hooks.py\", line 211, in _emit\n    response = handler(**kwargs)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\signers.py\", line 90, in handler\n    return self.sign(operation_name, request)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\signers.py\", line 160, in sign\n    auth.add_auth(request)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\site-packages\\botocore\\auth.py\", line 357, in add_auth\n    raise NoCredentialsError\nbotocore.exceptions.NoCredentialsError: Unable to locate credentials"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1585646353.011600",
        "client_msg_id": "d4450d1b-6bb7-4724-b95c-fc1fe54f289e",
        "text": "You need to configure your AWS credentials",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "Jq2",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "You need to configure your AWS credentials"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "U4H19D1D2",
        "type": "message",
        "ts": "1585646409.012600",
        "client_msg_id": "0c5d5185-8c84-47da-a31d-efa0dde4738a",
        "text": "just use the MarketRecorder if you want to save locally but I will be honest if you really need programming experience (basic is fine) to use these libraries. The errors are telling you the problem",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "137c5a3ef323",
            "image_72": "https:\/\/avatars.slack-edge.com\/2024-03-15\/6819395173841_137c5a3ef323f1944a1a_72.png",
            "first_name": "liam",
            "real_name": "liam",
            "display_name": "liam",
            "team": "T4G9NBD2M",
            "name": "liam",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "g=QEY",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "just use the MarketRecorder if you want to save locally but I will be honest if you really need programming experience (basic is fine) to use these libraries. The errors are telling you the problem"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "U010GM77S4W",
        "type": "message",
        "ts": "1585646647.013800",
        "client_msg_id": "c5543230-c67c-4987-a5d7-b5f46ffb9346",
        "text": "Ah I didn't realize I need AWS, how do I configure to save locally? (i am using marketrecorder). I do have some basic programming experience (VBA).",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "66cae8f5399c",
            "image_72": "https:\/\/avatars.slack-edge.com\/2020-03-20\/1004508356769_66cae8f5399cf91a07ae_72.jpg",
            "first_name": "",
            "real_name": "mandelbot",
            "display_name": "mandelbot",
            "team": "T4G9NBD2M",
            "name": "demhadbale",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "NC5X",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Ah I didn't realize I need AWS, how do I configure to save locally? (i am using marketrecorder). I do have some basic programming experience (VBA)."
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "U4H19D1D2",
        "type": "message",
        "ts": "1585646728.014100",
        "client_msg_id": "b5d174a8-2349-44d8-9af1-e40d65accc9f",
        "text": "replace the S3MarketRecorder with this one <https:\/\/github.com\/liampauling\/flumine\/blob\/2a261e27d5738ed405f9f1d24446bdfe8b078cc1\/examples\/strategies\/marketrecorder.py#L16>",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "137c5a3ef323",
            "image_72": "https:\/\/avatars.slack-edge.com\/2024-03-15\/6819395173841_137c5a3ef323f1944a1a_72.png",
            "first_name": "liam",
            "real_name": "liam",
            "display_name": "liam",
            "team": "T4G9NBD2M",
            "name": "liam",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "attachments": [
            {
                "service_name": "GitHub",
                "title": "liampauling\/flumine",
                "title_link": "https:\/\/github.com\/liampauling\/flumine\/blob\/2a261e27d5738ed405f9f1d24446bdfe8b078cc1\/examples\/strategies\/marketrecorder.py#L16",
                "text": "flūmine - Betfair trading framework. Contribute to liampauling\/flumine development by creating an account on GitHub.",
                "fallback": "GitHub: liampauling\/flumine",
                "thumb_url": "https:\/\/avatars2.githubusercontent.com\/u\/12016537?s=400&v=4",
                "from_url": "https:\/\/github.com\/liampauling\/flumine\/blob\/2a261e27d5738ed405f9f1d24446bdfe8b078cc1\/examples\/strategies\/marketrecorder.py#L16",
                "thumb_width": 392,
                "thumb_height": 392,
                "service_icon": "https:\/\/a.slack-edge.com\/80588\/img\/unfurl_icons\/github.png",
                "id": 1,
                "original_url": "https:\/\/github.com\/liampauling\/flumine\/blob\/2a261e27d5738ed405f9f1d24446bdfe8b078cc1\/examples\/strategies\/marketrecorder.py#L16"
            }
        ],
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "uNqj1",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "replace the S3MarketRecorder with this one "
                            },
                            {
                                "type": "link",
                                "url": "https:\/\/github.com\/liampauling\/flumine\/blob\/2a261e27d5738ed405f9f1d24446bdfe8b078cc1\/examples\/strategies\/marketrecorder.py#L16"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "U010GM77S4W",
        "type": "message",
        "ts": "1585646758.014400",
        "client_msg_id": "79525660-0085-4b82-91c7-f8c240617f1c",
        "text": "Thanks Liam",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "66cae8f5399c",
            "image_72": "https:\/\/avatars.slack-edge.com\/2020-03-20\/1004508356769_66cae8f5399cf91a07ae_72.jpg",
            "first_name": "",
            "real_name": "mandelbot",
            "display_name": "mandelbot",
            "team": "T4G9NBD2M",
            "name": "demhadbale",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "ecY",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Thanks Liam"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "subtype": "channel_join",
        "user": "U010RPB4S9Y",
        "text": "<@U010RPB4S9Y> has joined the channel",
        "type": "message",
        "ts": "1585665630.014600"
    },
    {
        "subtype": "channel_join",
        "user": "U01158P1UTX",
        "text": "<@U01158P1UTX> has joined the channel",
        "type": "message",
        "ts": "1585681108.014800"
    }
]